src/doc/trpl/glossary.md

```diff
@@ -0,0 +1,17 @@
+% Glossary
+
+In this section we briefly explain terms that you may not be familiar with.
+
+This book avoids, as much as possible, overcomplicated terms. Nevertheless, the following were deemed important enough that omitting them would be a disservice to the reader.
```

Comment:
I would prefer to be a bit more positive here. I'm not sure exactly what I'd want to write, but something more like "not everyone who comes to Rust has a background in systems programming, nor in computer science, so we've added explanations of domain terms that might be unfamilliar" or something